filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobFileSystemManager.scala

```diff
@@ -82,31 +84,26 @@ case class BlobFileSystemManager(
 sealed trait BlobTokenGenerator {def generateAccessToken: Try[AzureSasCredential]}
 object BlobTokenGenerator {
   def createBlobTokenGenerator(container: BlobContainerName, endpoint: EndpointURL, subscription: Option[SubscriptionId]): BlobTokenGenerator = {
-    createBlobTokenGenerator(container, endpoint, None, None, subscription)
+    NativeBlobTokenGenerator(container, endpoint, subscription)
   }
-  def createBlobTokenGenerator(container: BlobContainerName,
-                               endpoint: EndpointURL,
-                               workspaceId: Option[WorkspaceId],
-                               workspaceManagerURL: Option[WorkspaceManagerURL],
-                               subscription: Option[SubscriptionId]
-                              ): BlobTokenGenerator = {
-     (container: BlobContainerName, endpoint: EndpointURL, workspaceId, workspaceManagerURL) match {
-       case (container, endpoint, None, None) =>
-         NativeBlobTokenGenerator(container, endpoint, subscription)
-       case (container, endpoint, Some(workspaceId), Some(workspaceManagerURL)) =>
-         WSMBlobTokenGenerator(container, endpoint, workspaceId, workspaceManagerURL)
-       case _ =>
-         throw new Exception("Arguments provided do not match any available BlobTokenGenerator implementation.")
-     }
+  def createBlobTokenGenerator(container: BlobContainerName, endpoint: EndpointURL, workspaceId: WorkspaceId, workspaceManagerClient: WorkspaceManagerApiClientProvider): BlobTokenGenerator = {
+    WSMBlobTokenGenerator(container, endpoint, workspaceId, workspaceManagerClient)
   }
-  def createBlobTokenGenerator(container: BlobContainerName, endpoint: EndpointURL): BlobTokenGenerator = createBlobTokenGenerator(container, endpoint, None)
-  def createBlobTokenGenerator(container: BlobContainerName, endpoint: EndpointURL, workspaceId: Option[WorkspaceId], workspaceManagerURL: Option[WorkspaceManagerURL]): BlobTokenGenerator =
-      createBlobTokenGenerator(container, endpoint, workspaceId, workspaceManagerURL, None)
```

Comment:
I pared down this helper class to exactly two functions, one for each of `NativeBlobTokenGenerator` and `WSMBlobTokenGenerator`. The conditional logic promoted to where we interpret the config, it seemed better to get it out of the way as early as possible and not pass around `Option`. Can always tweak/revert.